modal/mount.py

```diff
@@ -324,6 +324,9 @@ def add_local_dir(
         condition: Optional[Callable[[str], bool]] = None,
         # add files from subdirectories as well
         recursive: bool = True,
+        # Predicate filter function for file selection, which should accept a filepath and return `True` for inclusion.
+        # Defaults to including all files.
+        ignore: Union[list[str], Callable[[Path], bool]] = [],
```

Comment:
Note that we are 3.9+ now
```suggestion
        ignore: list[str] | Callable[[Path], bool] = [],
```
will probably add pyupgrade as a pre-commit step at some point so you don't need to think about this (although it had a few annoyin edge cases, e.g. around `modal.Dict`.

Reply:
I think this was introduced in 3.10 actually, we use this syntax in some places but only when we use `from __future__ import annotations` which I would like to avoid

Reply:
oh waow you're right